backend/service/auditsink/slack/slack.go

```diff
@@ -38,6 +44,19 @@ func New(cfg *any.Any, logger *zap.Logger, scope tally.Scope) (service.Service,
 	return s, nil
 }
 
+// The struct is used in a Go template and the fields need to be exported
+type auditEventMetadata struct {
+	Request  map[string]interface{}
+	Response map[string]interface{}
+}
+
+// helper functions to format values in the Go template
+var funcMap = template.FuncMap{
+	// for inputs that are type slice/map, returns a formatted slack list OR `N/A` if the slice/map is empty
+	// currently only iterates 1-level deep
+	"slackList": slackList,
```

Comment:
kept this helper b/c I still think it's helpful for message fields that are simple slice/map types (ie `map<string, string>`, `repeated strings`, etc). This way users won't need to write a `range` for these kind of simple fields and helps cut down on long templating syntax in the config.